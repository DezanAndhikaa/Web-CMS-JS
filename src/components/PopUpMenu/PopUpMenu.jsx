import React from 'react';
import './PopUpMenu.scss';
import LogOut from '@material-ui/icons/PowerSettingsNew';
import Setting from '@material-ui/icons/BrightnessLow';
import HistoryIcon from '@material-ui/icons/History';
import { Menu, StorageKey } from '../../constants';
import { ListItemIcon, MenuItem, MenuList, Modal, Paper, Typography } from '@material-ui/core';
import LogoutModal from '../../views/Logout/Logout';

export default class PopUpMenu extends React.PureComponent {

  constructor(props) {
    super(props);
    this.state = {
      isLogoutModalShown: true
    };
  }

  handleLogout = () => {
    localStorage.removeItem(StorageKey.USER_DATA);
    this.props.logout();
    this.props.onLogout();
    this.setState({ isLogoutModalShown: false });
    this.props.push(Menu.LOGIN);
  };

  handleClick = (menu, tab) => {
    this.props.push({
      pathname: menu,
      whichTab: tab
    });
  }

  onClickOpenModalLogout = e => {

    this.setState({
      isLogoutModalShown: true
    })

    console.log(this.state.isLogoutModalShown);
  }

  onClickLogOut() {
    return (
      <LogoutModal
        {...this.props}
        open={this.state.isLogoutModalShown}
        onYesClicked={this.handleLogout}
        onNoClicked={() => this.setState({ isLogoutModalShown: false })}
      />
    )
  }

  render() {
    return (
      <Modal className="pop-up" open={this.props.openModal}>
        <Paper>
          <MenuList>
            <MenuItem onClick={() => this.handleClick(Menu.PLANNING_TRACKING_HISTORY)}>
              <ListItemIcon>
                <HistoryIcon />
              </ListItemIcon>
              <Typography variant="inherit">Tracking History</Typography>
            </MenuItem>
            <MenuItem>
              <ListItemIcon>
                <Setting />
              </ListItemIcon>
              <Typography variant="inherit">Setting</Typography>
            </MenuItem>
            <MenuItem onClick={this.onClickOpenModalLogout}>
              <ListItemIcon>
                <LogOut />
              </ListItemIcon>
              <Typography variant="inherit" noWrap>Log Out</Typography>
              <LogoutModal
                {...this.props}
                open={this.isLogoutModalShown}
                onYesClicked={this.handleLogout}
                onNoClicked={() => this.setState({ isLogoutModalShown: false })}
              />
            </MenuItem>
          </MenuList>
        </Paper>
      </ Modal>
    );
  }
}
