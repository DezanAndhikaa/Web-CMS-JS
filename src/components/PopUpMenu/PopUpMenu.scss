.pop-up{
    width: 20%;
    height: fit-content;
    margin: auto;
}