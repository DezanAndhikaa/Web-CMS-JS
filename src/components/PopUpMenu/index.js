import { connect } from 'react-redux';
import { push } from 'connected-react-router';
import PopUpMenu from './PopUpMenu';
import { removeDataAction } from 'core/StorageHelper';
import { USER_DATA } from '../../constants';
import { logoutAction } from 'actions/Planning';

const mapStateToProps = (state) => ({
	path: state.router.location.pathname,
});

const mapDispatchToProps = (dispatch) => ({
	push : (path) => dispatch(push(path)),
	logout: () => dispatch(removeDataAction(USER_DATA)),
	onLogout: () => dispatch(logoutAction()),
});

const userProfile = connect(mapStateToProps, mapDispatchToProps)(PopUpMenu);

export { userProfile as PopUpMenu };