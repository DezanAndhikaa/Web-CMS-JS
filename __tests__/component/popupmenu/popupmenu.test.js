import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import PopUpMenu from 'components/PopUpMenu/PopUpMenu';

Enzyme.configure({adapter:new Adapter()});
it("render pop up menu correctly", () => {
    const wrapper = shallow (
        <PopUpMenu />
    )
    expect(wrapper).toMatchSnapshot();
})