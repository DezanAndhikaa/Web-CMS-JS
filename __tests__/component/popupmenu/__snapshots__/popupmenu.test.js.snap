// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render pop up menu correctly 1`] = `
<ForwardRef(Modal)
  className="pop-up"
>
  <WithStyles(ForwardRef(Paper))
    className="paper-menu"
  >
    <ForwardRef(MenuList)
      className="menu-list"
    >
      <WithStyles(ForwardRef(MenuItem))
        className="menu-item1"
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(Typography))
          className="menu-label"
          variant="inherit"
        >
          Tracking History
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        className="menu-item2"
      >
        <WithStyles(ForwardRef(Typography))
          className="menu-label"
          variant="inherit"
        >
          Setting
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        className="menu-item3"
      >
        <WithStyles(ForwardRef(Typography))
          className="menu-label"
          noWrap={true}
          variant="inherit"
        >
          Log Out
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(MenuItem))>
    </ForwardRef(MenuList)>
  </WithStyles(ForwardRef(Paper))>
</ForwardRef(Modal)>
`;
